app/controllers/task_controller.rb

```diff
@@ -1,44 +0,0 @@
-class TaskController < ApplicationController
-  
-  def index # GET tasks/ show me a list of all tasks
-    @tasks = Task.all
-    
-  end
-
-  def show #GET one task/ show me this one particluar task, by iD
-    @task = Task.find(params[:id])
-
```

Comment:
You're getting a nil:nil class error for index and show because you're calling to a class/method that is undefined. Which probably means @task isn't being properly defined in show here so @task = Task.find(params[:id]) is returning nil.